compiler/src/dotty/tools/dotc/transform/localopt/FormatChecker.scala

```diff
@@ -0,0 +1,247 @@
+package dotty.tools.dotc
+package transform.localopt
+
+import scala.annotation.tailrec
+import scala.collection.mutable.{ListBuffer, Stack}
+import scala.reflect.{ClassTag, classTag}
+import scala.util.chaining.*
+import scala.util.matching.Regex.Match
+
+import java.util.{Calendar, Date, Formattable}
+
+import StringContextChecker.InterpolationReporter
+
+/** Formatter string checker. */
+abstract class FormatChecker(using reporter: InterpolationReporter):
```

Comment:
It's needed either to move it's body into existing `StringContextChecker` or remove `StringContextChecker`

Reply:
Removed.